Bluetooth: Audio: Fix and move assigned numbers

Thalley opened the pull request:
Fix the definitions of the audio-related assigned numbers and remove them from lc3.h

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/43098

Thalley committed:
Bluetooth: Audio: remove BT_HCI_CODING_FORMAT_LC3 from lc3.h

The BT_HCI_CODING_FORMAT_LC3 is defined in hci_types.h and
applications should use that value.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Remove LC3 from the assigned numbers defines

The defines from the Bluetooth Audio assigned numbers are not
specific for LC3. This commit removes the LC3 infix and
the reference to LC3 in the documentation for them.

It also cleans up some of the documentation and
names of the enums.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Move assigned numbers from lc3.h to audio.h

The assigned numbers are not specific to LC3, so they should be
defined in the more general audio.h header file.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

kartben commented on samples/bluetooth/broadcast_audio_sink/src/main.c:
```diff
@@ -108,8 +108,8 @@ static struct bt_conn *broadcast_assistant_conn;
 static const struct bt_audio_codec_cap codec_cap = BT_AUDIO_CODEC_CAP_LC3(
-	BT_AUDIO_CODEC_LC3_FREQ_16KHZ | BT_AUDIO_CODEC_LC3_FREQ_24KHZ,
-	BT_AUDIO_CODEC_LC3_DURATION_10, BT_AUDIO_CODEC_LC3_CHAN_COUNT_SUPPORT(1), 40u, 60u, 1u,
+	BT_AUDIO_CODEC_CAP_FREQ_16KHZ | BT_AUDIO_CODEC_CAP_FREQ_24KHZ,
```
looks like a breaking change or am I missing something? shouldn't the `...LC3...` option only be deprecated for now?
and would need entires in release notes / migration guide too

Thalley replied:
@kartben My understanding is that for experimental APIs (like LE Audio), this is not necessary. 

Given how long time these macros have existed, it may make sense to put something in the migration guide, but AFAIK that's not something we've done before for experimental APIs. 

What do you think?

kartben replied:
Ah, forgot that LE Audio was experimental, sorry. So yes, you're right, it's not strictly necessary. If you don't mind, and based on your comment re: how long they've been around, a short note in the migration guide would be welcome indeed. Thanks!

fredrikdanebjer reviewed (dismissed):
Aside from the comment from kartben above, lgtm

Thalley committed:
doc: releases: Add migration of the LC3 macro renames

Add a short migration guide for the LC3 macro renames.
Only a few examples are mentioned, as there are far too
many changes to describe them all.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

fredrikdanebjer commented on doc/releases/migration-guide-3.6.rst:
```diff
@@ -450,6 +450,14 @@ Bluetooth
+    more semantically correct name: e.g.
+    ``BT_AUDIO_CODEC_LC3_CHAN_COUNT`` is now ``BT_AUDIO_CODEC_CAP_TYPE_CHAN_COUNT``,
+    ``BT_AUDIO_CODEC_LC3_FREQ`` is now ``BT_AUDIO_CODEC_CAP_TYPE_FREQ``, and
+    ``BT_AUDIO_CODEC_CONFIG_LC3_FREQ`` is now ``BT_AUDIO_CODEC_CFG_FREQ``, etc. (:github:`67024`)
```
Some function and enum type names has also changed to used the more commonly used abbreviation of cfg and cap, e.g.:
bt_audio_codec_config_freq -> bt_audio_codec_cfg_freq
bt_audio_codec_capability_type -> bt_audio_codec_cap_type

Should you perhaps also mention this, as they also are part of public api?

Thalley replied:
Sure, I added some of the enums here as well (but far from all of them).

kartben commented on doc/releases/migration-guide-3.6.rst:
```diff
@@ -450,6 +450,18 @@ Bluetooth
 
+* Audio
+
+  * The BT_AUDIO_CODEC_LC3_* values from ``<zephyr/bluetooth/audio/lc3.h>`` have moved to
```
nit

```suggestion
  * The ``BT_AUDIO_CODEC_LC3_*`` values from ``<zephyr/bluetooth/audio/lc3.h>`` have moved to
```

Thalley replied:
Wasn't sure about that one, since it's not an actual value, but I added your suggestion anyways :)

kartben reviewed (commented):
migration guide lgtm, thanks! don't know enough about the topic to give an actual +1 though :)
